src/main/java/org/snomed/snowstorm/core/data/services/pojo/ConceptHistory.java

```diff
@@ -0,0 +1,43 @@
+package org.snomed.snowstorm.core.data.services.pojo;
+
+import org.snomed.snowstorm.core.data.domain.ComponentType;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * History of a Concept.
+ */
+public class ConceptHistory {
+    private final String conceptId;
+    private final Map<String, Set<ComponentType>> history = new HashMap<>();
+
+    public ConceptHistory() {
+        this.conceptId = null;
+    }
+
+    public ConceptHistory(String conceptId) {
+        this.conceptId = conceptId;
+    }
+
+    public String getConceptId() {
+        return this.conceptId;
+    }
+
+    public Map<String, Set<ComponentType>> getHistory() {
+        return this.history;
+    }
+
+    public void addToHistory(String effectiveTime, ComponentType componentType) {
+        Set<ComponentType> componentTypes = this.history.get(effectiveTime);
+        if (componentTypes != null) {
+            componentTypes.add(componentType);
+        } else {
+            componentTypes = new HashSet<>();
+            componentTypes.add(componentType);
+            this.history.put(effectiveTime, componentTypes);
+        }
```

Comment:
You could replace the whole contents of this method using Java 8 syntax, which is quicker to read and write once you get used to it:
```
history.computeIfAbsent(effectiveTime, (key) -> new HashSet<>()).add(componentType)
```
The `key` in this example is not used in the map value builder and can have any name.

Reply:
Very succinct; I have changed this.